lib/modules/dosomething/dosomething_reportback/dosomething_reportback.module

```diff
@@ -522,6 +520,32 @@ function dosomething_reportback_form_alter(&$form, &$form_state, $form_id) {
   }
 }
 
+/**
+ * Callback for /rbf/FID page.
+ */
+function dosomething_reportback_permalink_view($args) {
+  $rbid = $args[0];
+  $fid = $_GET['fid'];
+
+  if (variable_get('rogue_collection', FALSE)) {
+    return dosomething_reportback_view_rogue($rbid, $fid);
+  }
+
+  return dosomething_reportback_view_entity(entity_load('reportback', $rbid));
+}
+
+/**
+ * Callback for /rbf/FID page, when Rogue is enabled.
+ */
+function dosomething_reportback_view_rogue($rbid, $fid) {
+    // @TODO: Request signup from Rogue so we can display it on permalink page.
+    // $rogue_signup = dosomething_rogue_client()->get('v2/activity')
```

Comment:
We can easily add that in as a filter for sure! It just kinda kept snowballing into something bigger and adding more functionality to Rogue and going back and forth so thought it was better to do this now and then add later. 

Also, @DFurnes mentioned that it might be better to work on building out the Rogue API post launch with more RESTful API endpoints and this could be one of them `/signup/{signup_id}`.